src/main.cpp

```diff
@@ -7823,6 +7827,38 @@ bool SendMessages(const Consensus::Params& params, CNode* pto)
                 pto->fDisconnect = true;
             }
         }
+        // Check for headers sync timeouts
+        if (state.fSyncStarted && state.nHeadersSyncTimeout < std::numeric_limits<int64_t>::max()) {
+            // Detect whether this is a stalling initial-headers-sync peer
+            if (pindexBestHeader->GetBlockTime() <= GetTime() - 24*60*60) {
```

Comment:
What's the rationale for choosing 1 day here? I didn't see any on the Bitcoin PR. In any case it should be a named constant.

Reply:
I couldn't find any explanation for that 1 day either. My wild guess: it is an attempt to distinct stalled from slow peer. Maybe we shouldn't crack our heads trying to "over-engineer a guesstimate", too many unknown factors are at work (internet connection speed and stability, availability of peers, starting height of the current sync, etc...).

Reply:
This might be the explanation for "1 day" window: https://github.com/bitcoin/bitcoin/pull/10345#issuecomment-300147692

>  Once our headers chain is within 24 hours of current time, we'll request headers from our other peers as well, so after that point as long as one of your peers isn't stalling you, you'll get the headers chain.

Reply:
Zcash seems to be aligned with this according to https://github.com/zcash/zcash/blob/master/src/main.cpp#L7638-L7639

Reply:
Ah, I think that's it. I will push a commit to change the constant values and use a named constant for this one in both places.